scripts/spack.yaml

```diff
@@ -0,0 +1,37 @@
+spack:
+  definitions:
+    - platspecs: []
+    - when: 'platform == "linux"'
+      platspecs:
+      - valgrind
+  specs: 
+    - cmake
+    - doxygen
+    - geant4@11 cxxstd=17
+    - git
+    - git-lfs
+    - googletest
+    - hepmc3
+    - ninja
+    - nlohmann-json
+    - mpi
+    - python
+    - py-breathe
+    - py-sphinx
+    - py-sphinxcontrib-bibtex
+    - root cxxstd=17
+    - swig
+    - vecgeom +gdml cxxstd=17
+    - $platspecs
+  view: true
+  concretization: together
+  packages:
+    root:
+      variants: ~aqua ~davix ~examples ~opengl ~x ~tbb ~math ~gsl
+    all:
+      providers:
+        blas: [openblas]
+        lapack: [openblas]
+        mpi: [openmpi]
+      variants: cxxstd=17
```

Comment:
Does this render the explicit `cxxstd=17` definitions for Geant4, root and vecgeom redundant?

Reply:
They don't serve quite the same purpose: the `specs` are required and will fail if those variants don't exist. The variants under `packages` are "preferences" and are ignored if not applicable. Further complicating this process is that spack combines the two (in addition to site preferences from `packages.yaml`) in surprising ways...

Reply:
Ah gotcha, makes sense!